pytorch_lightning/core/datamodule.py

```diff
@@ -423,6 +423,12 @@ def __new__(cls, *args: Any, **kwargs: Any) -> "LightningDataModule":
         obj.prepare_data = cls._track_data_hook_calls(obj, obj.prepare_data)
         obj.setup = cls._track_data_hook_calls(obj, obj.setup)
         obj.teardown = cls._track_data_hook_calls(obj, obj.teardown)
+
+        # calling init method of LightningDataModule when its
```

Comment:
Mind improving this comment right there, so readers who don't know abut mro understand what is happening there ?

Reply:
Would this be fine?